src/store/RelayRecordStore.js

```diff
@@ -516,7 +516,7 @@ class RelayRecordStore {
         dataID
       );
       return item.__dataID__;
-    });
+    }).filter(id => this.getRecordState(id) !== 'NONEXISTENT');
```

Comment:
Sorry for the delay in reviewing this. The logic here is exactly what we want, but this will end up allocating an extra array. Maybe allocate a new array and use `forEach` instead of map + filter?